frontend/lib/src/components/elements/DeckGlJsonChart/useDeckGl.tsx

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useCallback, useEffect, useMemo, useState } from "react"
+
+import JSON5 from "json5"
+import {
+  CartoLayer,
+  colorBins,
+  colorCategories,
+  colorContinuous,
+} from "@deck.gl/carto/typed"
+import * as layers from "@deck.gl/layers/typed"
+import { JSONConverter } from "@deck.gl/json/typed"
+import * as geoLayers from "@deck.gl/geo-layers/typed"
+import * as aggregationLayers from "@deck.gl/aggregation-layers/typed"
+import * as meshLayers from "@deck.gl/mesh-layers/typed"
+import { PickingInfo } from "@deck.gl/core/typed"
+import isEqual from "lodash/isEqual"
+import { ViewStateChangeParameters } from "@deck.gl/core/typed/controllers/controller"
+import { TooltipContent } from "@deck.gl/core/typed/lib/tooltip"
+
+import type { DeckObject, PropsWithHeight, StreamlitDeckProps } from "./types"
+
+export type UseDeckGlShape = {
+  createTooltip: (info: PickingInfo | null) => TooltipContent
+  deck: DeckObject
+  onViewStateChange: (params: ViewStateChangeParameters) => void
+  viewState: Record<string, unknown>
+}
+
+export type UseDeckGlProps = Omit<PropsWithHeight, "theme" | "mapboxToken"> & {
+  isLightTheme: boolean
+}
+
+const DEFAULT_DECK_GL_HEIGHT = 500
+
+const configuration = {
+  classes: {
+    ...layers,
+    ...aggregationLayers,
+    ...geoLayers,
+    ...meshLayers,
+    CartoLayer,
+  },
+  functions: {
+    colorBins,
+    colorCategories,
+    colorContinuous,
+  },
+}
+
+export const jsonConverter = new JSONConverter({ configuration })
+
+export const interpolate = (info: PickingInfo, body: string): string => {
```

Comment:
nit: maybe add a quick comment on what this function does

Reply:
FWIW, this is 100% existing code that was just moved from the class component: https://github.com/streamlit/streamlit/blob/61fe8289c82a8ab80175ed9cef0df83857970988/frontend/lib/src/components/elements/DeckGlJsonChart/DeckGlJsonChart.tsx#L230-L247 but I will add a JSDoc comment!